java/arcs/core/host/EntityHandleManager.kt

```diff
@@ -40,9 +40,11 @@ import arcs.core.storage.handle.SingletonHandle
  */
 class EntityHandleManager(
     private val handleManager: HandleManager,
-    private val arcId: String = Id.Generator.newSession().newArcId("arc").toString(),
-    private val hostId: String = "nohost"
+    private val arcName: String = "defaultArcName",
+    private val hostId: String = "nohost",
+    private val idGenerator: Id.Generator = Id.Generator.newSession()
 ) {
+    private val arcId: String = idGenerator.newArcId(arcName).toString()
```

Comment:
The arcId comes from the Allocator and send to the ArcHost via Plan.Partition. All ArcHosts share this ArcId. I don't think we should be generating arcIds.  

The previous code had defaults for these so it could be used independent of an ArcHost, but normally, the ArcHost would hand the arcId to the EntityHandleManager.

Reply:
It might now be possible to share a generator session Per Arc, but only per Arc per ArcHost, otherwise, we need to add the currentSessionId to the Plan.Partition so that the allocator can share it amongst all participating ArcHosts. 

the method entityHandleManager() in AbstractArcHost should be passing arcId. If you want to keep around a Id.Generator, I'd recommend adding it to ArcHostContext, so that there's 1 per arcId per arcHost, and then pass arcHostContext.idGenerator to entityHandleManager() method.